backend/src/main/java/com/google/sps/servlets/SearchMatchServlet.java

```diff
@@ -116,10 +128,16 @@ private boolean isExpired(Participant participant) {
   }
 
   /** Send JSON response for expired participant that has been removed from datastore */
-  private void sendExpiredResponse(HttpServletResponse response) throws IOException {
+  private void sendExpiredResponse(HttpServletResponse response, Participant participant)
+      throws IOException {
     JSONObject expired = new JSONObject();
     expired.put(JSON_MATCH_STATUS, "expired");
-
+    try {
+      emailNotifier.sendExpiredEmail(
+          participant.getUsername(), userService.getCurrentUser().getEmail());
```

Comment:
Passing participant in as a parameter passes unnecessary info. You could pass just the username (not participant.getUsername(), but the String username from line 75) and either append the "@google.com" when passing email to sendExpiredEmail here OR just pass the username and append the "@google.com" in the sendExpiredEmail() method in EmailNotifier

Reply:
Kevin had recommended against adding @google.com to the strings . Also in the case in which i get the username form line 75 should username be set to a class field ?

Reply:
dhruvi mentioned extracting the logic form line 70-75 so I will try to go about doing that !

Reply:
What if we added a getUsernameFromEmail method that would take in the email which we retrieve from the userService here and just returns `email.split("@")[0]`? Then, I think we would only need the email param since the username can be retrieved from the email

Reply:
@kmh287

Reply:
The email notifier should take in the full email address rather than construct it internally from a username + "@google.com" which assumes the user's email domain. `email` in the calling method has the email address, so how about passing that instead of the participant

Reply:
I have added a getter method similar to that of addPartcipantServlet and call that instead of taking a participant in the constructor